src/index.ts

```diff
@@ -31,25 +34,34 @@ const mapUrlWithParsedQuery = (fn: MapUrlWithParsedQueryFn) =>
         urlHelpers.format,
     );
 
+const replaceQueryInParsedUrl = ({
+    newQuery,
+}: {
+    newQuery: BinaryUpdate<UrlWithParsedQuery['query'], ParsedUrlQueryInput>;
+}): MapUrlWithParsedQueryFn => ({ parsedUrl }) => ({
```

Comment:
Whilst we're here, I wanted to bring something up which I've been thinking about for awhile.

Currently we use named parameters for these functions. However, I don't think it's really necessary to do that, and I'm worried this is putting people off from using this library.

WDYT about removing usage of named parameters?

```diff
-replaceQueryInUrl({ newQuery })({ parsedUrl })
+replaceQueryInUrl(newQuery)(parsedUrl)
```

Reply:
I totally agree; parameter bags make the API a bit awkward since these are single-responsibility functions with a very narrow focus, therefore there is no room for parameters.

(It may be safe to avoid the term _named parameters_ since JavaScript doesn't have anything like that; parameter bags simulate their behavior.)